### DISCUSSION TITLE
A Solax API option?

### DISCUSSION THREAD

**Discussion by Kaotic3** [wills106/homeassistant-solax-modbus#720] [Ideas]
Just out of curiosity - I am reading through all the technical details, the buying of products, the wiring them in, the configuring them to speak to your router - all to get the data into HA.

But Solax has an API - you can call to it and get data.  Is there a reason we don't have that as an option?

**Comment by wills106** [comment]
The API only works if you have a Pocket WiFi or possibly Lan.
Older Pocket WiFi's have a very low refresh rate of 5mins (or is it 10mins)

I only have a Modbus connection over TCP, so that's why I created a Modbus Integration and started to widen it to other brand Inverters which use Modbus.

**Reply by Kaotic3** [reply]
You are right, it is only 5 minutes - but realistically it is about historic data isn't it? Rather than real-time by the second data?  Yes, it is nice to see what is going in and out at any given moment - but it isn't critical for this kind of tool.

Ok, I see this is in python - if I were able to write up a separate API function, would it be possible for you to wire those results into your overall application?

This would require creating a page for the API key perhaps?

I created an API app awhile ago because I have 2 batteries and Solax does a really bad job of recording data across two batteries and I wasn't sure both were working - so I built an app that recorded the input totals so that I could get a numeric value rather than a % what was stored....

It was a while ago and I am better at C# than I am python, but I can give it a shot?
